app/src/androidTest/java/com/github/se/signify/ui/screens/home/QuestScreenTest.kt

```diff
@@ -1,64 +1,188 @@
-import androidx.compose.ui.test.onNodeWithTag
 import com.github.se.signify.model.quest.Quest
 import com.github.se.signify.model.quest.QuestRepository
 import com.github.se.signify.model.quest.QuestViewModel
+import com.github.se.signify.model.user.UserRepository
+import com.github.se.signify.model.user.UserViewModel
 import com.github.se.signify.ui.navigation.NavigationActions
 import com.github.se.signify.ui.navigation.Route
 import com.github.se.signify.ui.screens.home.QuestBox
+import com.github.se.signify.ui.screens.home.QuestDescriptionDialog
 import com.github.se.signify.ui.screens.home.QuestScreen
 import org.junit.Before
 import org.junit.Rule
 import org.junit.Test
 import org.mockito.Mockito.mock
+import org.mockito.Mockito.verify
 import org.mockito.Mockito.`when`
 
 class QuestScreenTest {
 
   private lateinit var questRepository: QuestRepository
+  private lateinit var userRepository: UserRepository
   private lateinit var navigationActions: NavigationActions
   private lateinit var questViewModel: QuestViewModel
+  private lateinit var userViewModel: UserViewModel
 
-  val sampleQuest =
+  private val sampleQuest =
       Quest(index = "1", title = "Sample Quest", description = "This is a sample quest description")
 
   @get:Rule val composeTestRule = createComposeRule()
 
   @Before
   fun setUp() {
     questRepository = mock(QuestRepository::class.java)
+    userRepository = mock(UserRepository::class.java)
     navigationActions = mock(NavigationActions::class.java)
     questViewModel = QuestViewModel(questRepository)
+    userViewModel = UserViewModel(userRepository)
 
     `when`(navigationActions.currentRoute()).thenReturn(Route.QUEST)
   }
 
   @Test
   fun hasRequiredComponent() {
     composeTestRule.setContent {
-      QuestScreen(navigationActions = navigationActions, questViewModel = questViewModel)
+      QuestScreen(
+          navigationActions = navigationActions,
+          questViewModel = questViewModel,
+          userViewModel = userViewModel)
     }
 
     composeTestRule.onNodeWithTag("QuestScreen").assertIsDisplayed()
   }
 
   @Test
   fun questBoxDisplaysCorrectInformation() {
-    composeTestRule.setContent { QuestBox(quest = sampleQuest) }
+    composeTestRule.setContent { QuestBox(quest = sampleQuest, isUnlocked = true) }
 
     composeTestRule.onNodeWithTag("QuestCard").assertIsDisplayed()
     composeTestRule.onNodeWithTag("QuestHeader").assertIsDisplayed()
-    composeTestRule.onNodeWithTag("QuestTitle").assertIsDisplayed()
-    composeTestRule.onNodeWithTag("QuestDescription").assertIsDisplayed()
     composeTestRule.onNodeWithTag("QuestActionButton").assertIsDisplayed()
   }
 
   @Test
   fun questActionButtonHasClickAction() {
-    composeTestRule.setContent { QuestBox(quest = sampleQuest) }
+    composeTestRule.setContent { QuestBox(quest = sampleQuest, isUnlocked = true) }
 
     composeTestRule.onNodeWithTag("QuestActionButton").assertHasClickAction()
   }
+
```

Comment:
The tests added are consistent with the UI,and the coverage is higher. Good job on that